app/src/main/java/com/randomappsinc/studentpicker/editing/DuplicationDialog.java

```diff
@@ -11,35 +11,41 @@
 public class DuplicationDialog {
 
     public interface Listener {
-        void onDuplicationSubmitted(String name, int amountToAdd);
+        void onDuplicationSubmitted(String name, int amountToHave, int currentAmount);
     }
 
     private MaterialDialog dialog;
     private String currentName;
+    private int currentAmount;
 
     public DuplicationDialog(Context context, final Listener listener) {
         dialog = new MaterialDialog.Builder(context)
                 .content(R.string.cloning_title)
                 .inputType(InputType.TYPE_CLASS_NUMBER)
                 .input(context.getString(R.string.num_copies), "", (dialog, input) -> {
-                    boolean isValid = input.length() > 0 && (Integer.parseInt(input.toString()) > 0);
-                    dialog.getActionButton(DialogAction.POSITIVE).setEnabled(isValid);
+                    if (input.length() > 0) {
```

Comment:
I think this code breaks the case where you type in a valid input and then delete it back to nothing (the button will stay enabled). If input.length() is 0, we aren't mutating the dialog's positive action button.

Reply:
You're right, I didn't notice that case